### ISSUE TITLE
FR: Support workload identity instead of auth keys?

### ISSUE THREAD

**Issue by arianvp** [tailscale/tailscale#13087]
### What are you trying to do?

Tailscale's whole selling point is that they don't want my passwords. They achieve this through OIDC and works beautifully for humans. However for servers the best tailscale has to offer is that I need to handle a private auth key that I should keep secure and securely transfer to my server. This is a pain. Tailscale does seem to want my passwords.

I want to find a way for my servers to join without some kind of bearer token needing to be pre-generated

### How should we solve this?

Google Cloud, Kubernetes and GitHub Actions all use OIDC for workload identity. Allow machines to authenticatie with ID tokens just like humans!

https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/about-security-hardening-with-openid-connect

### What is the impact of not solving this?

No more need for needing to provision auth keys and manage sensitive secrets.

**Comment by arianvp** [comment]
IAM Role support would also be very appreciated

**Comment by rmb938** [comment]
I would also like this as a supported auth method.

For example GCP supports Github Actions as an identity provider https://github.com/google-github-actions/auth?tab=readme-ov-file#direct-wif

As well as hasicorp vault https://github.com/hashicorp/vault-action?tab=readme-ov-file#jwt-with-github-oidc-tokens

This could be very useful when using Tailscale within Github actions, instead of needing to provide a OAuth or API Keys Tailscale could support Github Actions OIDC login.

This could also be extended to other systems like allowing agents to auth with clloud service accounts and whatnot.

**Comment by arianvp** [comment]
Is this implemented now? https://tailscale.com/blog/workload-identity-beta

**Comment by mcoulombe** [comment]
Yup this is implemented, with more improvements to come as we prepare the GA release. Clients are not forced to rely on some manually transferred secrets such as an authkey or client secret to join a tailnet anymore.

**Comment by arianvp** [comment]
Great news. Thanks for listening to the customer feedback! :D

**Comment by mpminardi** [comment]
Closing this out since this has been released! See https://tailscale.com/kb/1581/workload-identity-federation for further docs here.
